### ISSUE TITLE
MVDR module can't work in distributed training

### ISSUE THREAD

**Issue by nateanl** [pytorch/audio#1991]
### 🐛 Describe the bug

NCCL currently only supports floating point and integer dtype. (See https://github.com/pytorch/pytorch/issues/45760)
The MVDR module forces to use ``cdouble`` dtype to improve the numerical robustness, however, it will cause an runtime error when integrating it to a neural network:
```python
NCCL Backend doesn't support torch.complexdouble data type
```
To solve it, the temporary workaround is relax the dtype constraint by allowing MVDR to use ``cfloat`` dtype.

### Versions

The issue is general to the current PyTorch and torchaudio version.

**Comment by mthrok** [comment]
This makes me wonder ... Should we keep using pseudo complex type? #1337

**Comment by nateanl** [comment]
I tested ``view_as_complex``, it actually has the same functionality. But I made it work by converting the waveform to ``torch.float`` after MVDR beamforming.
So the current MVDR indeed works in distributed training, just need to convert the output to floating point precision. I'm wondering how to check the dtype of gradient in MVDR to make sure it's double instead of float?

**Comment by nateanl** [comment]
The PyTorch that works is built from source. Let me try with a stable version to see how it is like.

**Comment by nateanl** [comment]
I think this issue is solved in latest PyTorch. Using ``torch.cdouble`` inside MVDR works fine within distributed training. I only need to convert it to ``torch.float``after InverseSpectrogram.

After discussion with @fmassa, I'm thinking adding ``torch.cfloat`` is also a good option, if the performance is not degraded. I will verify it by using ``torch.cfloat`` in the tutorial.

**Comment by nateanl** [comment]
Close it since MVDR works in latest PyTorch in distributed training.
